### PR TITLE
feat(ListItemExandIcon): added new list component for expanded icon view

### PR DESCRIPTION
**Pull request by IsaevAlexandr**
- fixed expanded icon position
- expanded list icon separated by own component to reuse
- added examples in storys and docs

### CODE REVIEW

File: src/components/useList/components/ListItemExpandIcon/ListItemExpandIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                colorText({color: disabled ? 'hint' : undefined}),
+            )}
+            data={ChevronDown}
+            size={16}
+        />
+    );
+};
+
+// For correct rendering inside `Button` component
+ListItemExpandIcon.displayName = 'Icon';
```
Comment (amje) [inline src/components/useList/components/ListItemExpandIcon/ListItemExpandIcon.tsx:35]: For correct rendering there is a `Button.Icon` wrapper, everything placed inside would work as an icon

File: src/components/useList/components/ListItemExpandIcon/__stories__/list-item-expand-icon.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+<ListItemExpandIconInsideButton />
+
+#### Props
+
+| Name              | Description                                     |                         Type                         | Default |
+| :---------------- | :---------------------------------------------- | :--------------------------------------------------: | :-----: |
+| expanded          | icon state                                      |                      `boolean`                       |         |
+| disableTransition | disable animation while `expanded` state change |                      `boolean`                       |         |
```
Comment (amje) [inline src/components/useList/components/ListItemExpandIcon/__stories__/list-item-expand-icon.md:45]: Why we need to disable the animatiom?

File: src/components/useList/components/ListItemExpandIcon/ListItemExpandIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+export interface ListItemExpandIconProps extends ListItemExpandIconRenderProps {}
+
+export const ListItemExpandIcon = ({
+    expanded,
+    disableTransition,
+    usageRole: usage = 'action',
+    disabled,
+}: ListItemExpandIconProps) => {
+    return (
+        <Icon
```
Comment (amje) [inline src/components/useList/components/ListItemExpandIcon/ListItemExpandIcon.tsx:23]: Is there a chance we could reuse `ArrowToggle` component here?

File: src/components/useList/components/ListItemView/ListItemViewContent.tsx [hunk 4]
```diff
@@ -57,7 +58,17 @@ export const ListItemViewContent = ({
     expanded,
     selected,
     title,
+    expandIconPosition = 'start',
```
Comment (amje) [inline src/components/useList/components/ListItemView/ListItemViewContent.tsx:61]: `expandIconPlacement`

File: src/components/useList/components/ListItemView/ListItemViewContent.tsx [hunk 5]
```diff
@@ -57,7 +58,17 @@ export const ListItemViewContent = ({
...
+    renderExpandIcon: RenderExpandIcon = ListItemExpandIcon,
```
Comment (amje) [inline src/components/useList/components/ListItemView/ListItemViewContent.tsx:62]: What the case for custom render?

File: src/components/useList/components/ListItemExpandIcon/ListItemExpandIcon.tsx (hunk 1 above)
Comment (IsaevAlexandr) [reply src/components/useList/components/ListItemExpandIcon/ListItemExpandIcon.tsx:35]: This is an undocumented property (`Button.Icon`), how will users know about it?

File: src/components/useList/components/ListItemView/ListItemViewContent.tsx (hunk 5 above)
Comment (IsaevAlexandr) [reply src/components/useList/components/ListItemView/ListItemViewContent.tsx:62]: if you wont custom icon view

Comment (amje) [reply src/components/useList/components/ListItemView/ListItemViewContent.tsx:62]: This prop called `render*` but it's not a function. Also it takes props of `ListItemExpandIcon`, so when I want custom, I should rely on this props?. Feels a bit overwhelming and hard to understand.

File: src/components/useList/components/ListItemExpandIcon/__stories__/list-item-expand-icon.md (hunk 2 above)
Comment (amje) [reply src/components/useList/components/ListItemExpandIcon/__stories__/list-item-expand-icon.md]: It's our design decision to have transitions on such arrows. Look at the `ArrowToggle`, we don't have it. Allowing to disable it here would feels like something is broken...

File: src/components/useList/components/ListItemView/ListItemViewContent.tsx (hunk 5 above)
Comment (IsaevAlexandr) [reply src/components/useList/components/ListItemView/ListItemViewContent.tsx:62]: this is exactly a render function with props. Here fased link to a `ListItemExpandIcon` withot calling. In most cases you don't need to use this prop - it's optional

**Review by amje — APPROVED** [review]

**Pull request merged**